pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/creator/impl/SegmentColumnarIndexCreator.java

```diff
@@ -196,6 +204,15 @@ public void init(SegmentGeneratorConfig segmentCreationSpec, SegmentIndexCreatio
       Preconditions.checkState(dictEnabledColumn || !invertedIndexColumns.contains(columnName),
           "Cannot create inverted index for raw index column: %s", columnName);
 
+      boolean forwardIndexDisabled = forwardIndexDisabledColumns.contains(columnName);
+      if (forwardIndexDisabled) {
+        Preconditions.checkState(dictEnabledColumn && invertedIndexColumns.contains(columnName)
+            && !columnIndexCreationInfo.isSorted() && (!rangeIndexColumns.contains(columnName)
+                || rangeIndexVersion == BitSlicedRangeIndexCreator.VERSION), String.format("Cannot disable forward "
+            + "index for column: %s without dictionary and inverted index or which is sorted or which has range "
+            + "index with version < 2", columnName));
```

Comment:
(nit) may be cleaner to split this into 2-3 conditions. Will also make it clear to user what exact combination is invalid. Consider writing a simple helper method and fold everything into it

Reply:
good point, done!